offgridplanner/optimization/processing.py

```diff
@@ -237,13 +237,40 @@ def collect_supply_opt_json_data(self):
         return supply_opt_json
 
     def collect_grid_opt_json_data(self):
-        # calculate periodical costs of components out of input capex, opex and lifetime
+        nodes_df = self.project.nodes.df.copy()
+        nodes_df = nodes_df.reset_index(drop=False)
+        nodes_df = nodes_df.rename(columns={"index": "id"})
+        nodes_records = nodes_df.to_dict(orient="records")
+
+        node_fields = [
+            "id",
+            "how_added",
+            "node_type",
+            "consumer_type",
+            "custom_specification",
+            "shs_options",
+            "consumer_detail",
+            "is_connected",
+            "coordinates",
+        ]
+
+        for node in nodes_records:
+            if "latitude" in node and "longitude" in node:
+                node["coordinates"] = [node.pop("latitude"), node.pop("longitude")]
+            else:
+                node["coordinates"] = None
+
+        nodes_values = [
+            [node.get(field, None) for field in node_fields] for node in nodes_records
+        ]
 
         grid_opt_json = {
-            "nodes": self.project.nodes.df.to_json(),
+            "node_fields": node_fields,
```

Comment:
This way of putting the data is more efficient as before. I compared the size and it was 2/3 of the previous one.